crates/project/src/project.rs

```diff
@@ -11466,19 +11467,21 @@ impl ProjectLspAdapterDelegate {
         worktree: &Model<Worktree>,
         cx: &ModelContext<Project>,
     ) -> Arc<Self> {
+        let load_direnv = ProjectSettings::get_global(cx).load_direnv.clone();
         Arc::new(Self {
             project: cx.weak_model(),
             worktree: worktree.read(cx).snapshot(),
             fs: project.fs.clone(),
             http_client: project.client.http_client(),
             language_registry: project.languages.clone(),
             shell_env: Default::default(),
+            load_direnv,
         })
     }
 
-    async fn load_shell_env(&self) {
+    async fn load_shell_env(&self, load_direnv: &DirenvSettings) {
```

Comment:
Why pass this in? Can't `load_shell_env` access `self` itself?

Reply:
Yeah, sorry for that, I'll fix it in a moment